### DISCUSSION TITLE
请问如何实现TextBox/RichTextBox同一行显示不同颜色的文字

### DISCUSSION THREAD

**Discussion by zeratulag** [BYJRK/DotNet-Discussions#76] [WPF 开发]
您好，我希望在TextBox（或者RichTextBox）中显示如下效果，其中的文字是可变的（Bing 到ViewModel）
![image](https://user-images.githubusercontent.com/59944417/230705610-a37231da-3511-4452-a49d-fd1ad6633a5d.png)

请问如何实现比较方便？

**Comment by zeratulag** [comment]
我尝试了用 RichTextBox 的 FlowDocument 属性表示，如下：

        <RichTextBox Grid.Row="0" Grid.Column="1" Style="{StaticResource Style.RichTextBlockBox}"
                     MaxWidth="400">
            <FlowDocument>
                <Paragraph Typography.Kerning="True">
                    <Run Foreground="Green"
                         Text="这是一行文字" />
                    <Run Foreground="Red"
                         Text="（这是不同颜色）"/>
                    <Run Foreground="White"
                         Text="文字结束"/>
                </Paragraph>
            </FlowDocument>
        </RichTextBox>

但是文字<Run>段落中间出现了间隙，很难看，请问有没有消除间隙的方法？
![image](https://user-images.githubusercontent.com/59944417/230709754-e75d1dae-f578-460f-a391-02156e82c424.png)

**Reply by PQGit** [reply]
把“><”符号间的空白删掉就好啦。

`<Run Foreground="Green" Text="这是一行文字" /><Run Foreground="Red" Text="（这是不同颜色）"/><Run Foreground="White" Text="文字结束"/>`

**Reply by zeratulag** [reply]
已解决，感谢！没想到是VS自动格式化代码的锅
